autosklearn/pipeline/components/data_preprocessing/text_encoding/bag_of_word_encoding_distinct.py

```diff
@@ -61,13 +63,18 @@ def transform(self, X: PIPELINE_DATA_DTYPE) -> PIPELINE_DATA_DTYPE:
         X_new = None
         if self.preprocessor is None:
             raise NotImplementedError()
-        # iterate over the pretrained preprocessors and columns and transform the data
-        for preprocessor, feature in zip(self.preprocessor, X.columns):
-            if X_new is None:
-                # possiblity to add TruncatedSVD here
-                X_new = preprocessor.transform(X[feature])
+
+        for key_preprocessor, feature in zip(sorted(self.preprocessor.keys()), sorted(X.columns)):
```

Comment:
This is a weird line. Why not simply go over all preprocessors and retrieve the correct column from `X`?